### PR TITLE
Polish expire options

### PR DESCRIPTION
**Pull request by Chepelau**
Basically only simple edits, also moved setting pool from adjust_lpool_balance_and_pool_locked_capital_expired_options to expire_opt_for_pool, also please check line 726 in liq_pool.cairo. Should that check be uncommented or removed(do we care if there is little position left)?

### CODE REVIEW

File: contracts/core_amm/liquidity_pool.cairo [hunk 1]
```diff
@@ -561,16 +561,6 @@ func adjust_lpool_balance_and_pool_locked_capital_expired_options{
...
-    let (current_pool_position) = get_option_position(
-        lptoken_address, option_side, maturity, strike_price
-    );
-    let current_pool_position_uint256: Uint256 = toUint256_balance(current_pool_position, lpool_underlying_token);
-
-    let new_pool_position = current_pool_position - option_size;
-    with_attr error_message("new_pool_position is negative in adjust_lpool_balance_and_pool_locked_capital_expired_options") {
-        assert_nn(new_pool_position);
-    }
-    set_option_position(lptoken_address, option_side, maturity, strike_price, new_pool_position);
```
Comment (tensojka) [inline contracts/core_amm/liquidity_pool.cairo:573]: the call to set_option_position was removed and isn't reintroduced elsewhere in this function, I understand that it was called twice before?

File: contracts/core_amm/liquidity_pool.cairo [hunk 2]
```diff
@@ -700,23 +698,33 @@ func expire_option_token_for_pool{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*
...
+        );
     }
 
+    let (current_pool_position) = get_option_position(
+        lptoken_address, option_side, maturity, strike_price
+    );
+
+    let new_pool_position = current_pool_position - option_size;
+    with_attr error_message("new_pool_position is negative in expire_option_token_for_pool") {
+        assert_nn(new_pool_position);
```
Comment (tensojka) [inline contracts/core_amm/liquidity_pool.cairo:724]: can we make this assert_le? this shouldn't be a problem with felts, but just to be overly cautious?

**Review by tensojka — CHANGES_REQUESTED** [review]

File: contracts/core_amm/liquidity_pool.cairo (hunk 1 above)
Comment (Chepelau) [reply contracts/core_amm/liquidity_pool.cairo:573]: it was called twice -  in this function, and right after that in the expire_option_token_for_pool with `new_pool_pos = 0`

Comment (Chepelau) [reply contracts/core_amm/liquidity_pool.cairo:573]: so the call in expire_option_token_for_pool would overwrite it anyway

File: contracts/core_amm/liquidity_pool.cairo (hunk 2 above)
Comment (Chepelau) [reply contracts/core_amm/liquidity_pool.cairo:724]: you mean assert_le(0, stuff)?

File: contracts/core_amm/liquidity_pool.cairo (hunk 1 above)
Comment (tensojka) [reply contracts/core_amm/liquidity_pool.cairo:573]: as long as you think it's tested enough (I believe pool position is tested after expiry) then LGTM

File: contracts/core_amm/liquidity_pool.cairo (hunk 2 above)
Comment (tensojka) [reply contracts/core_amm/liquidity_pool.cairo:724]: assert_le(option_size, current_pool_position)

**Review by tensojka — APPROVED** [review]

**Pull request merged**